Auth middleware

saralein opened the pull request.

saralein committed:
Rename authorizer to auth middleware

saralein committed:
Convert authorizer to auth middleware

zdavy commented on server/src/main/java/com/saralein/server/middleware/AuthMiddleware.java:
```diff
@@ -0,0 +1,63 @@
+        String encodedAuthorization = request.getHeader("Authorization");
+        String parsedAuthorization = parseRequestAuthorization(encodedAuthorization);
+        return !routes.requiresAuthorization(uri) || encodeValidAuthorization().equals(parsedAuthorization);
+    }
```
I think this is 90% of the way there, but I think this part feels like domain leak. I think it's strange for the routes to know about authorization. FWIW I don't think there's anything weird about the middleware getting a list of routes that need authorization, rather than all routes.

damonkelley replied:
Also, since this is a callable, and the controller is a callable, could you apply the middleware directly to the controller inside when you add it to the routes? 

That way, it would still be middleware, but it would be applied only for that controller.

saralein replied:
OK, I can adjust.  I remember having a list of routes a while ago, and I think there was a suggestion about somehow using the routes.  I could have also misconstrued the comment.  I seem to do that too frequently.

damonkelley replied:
We can do either/or. I just wanted to point out that option.

saralein replied:
I didn't make the controllers callable, but I could easily do that and wish I had thought of it. Might fix some of these issues and then do it on another branch just for all the files it will change.

damonkelley commented on server/src/main/java/com/saralein/server/middleware/AuthMiddleware.java:
```diff
@@ -0,0 +1,63 @@
+    private final String realm;
+    private Callable next;
+
+    public AuthMiddleware(Routes routes, String username, String password, String realm) {
```
I think this class is probably doing too much. I feel like that actual authentication should be handled in another object, and the middleware should use that. Yeah?

saralein committed:
Adjust auth middleware

- Abstract authorizer from auth middleware
- Remove auth list from routes
- Provide route to auth middleware

saralein committed:
Update controller to extend callable

saralein committed:
Adjust authorizer and auth middleware

- Change authorizer isAuthorized signature
- Wrap auth middleware around log controller
- Pass wrapped callable to routes

saralein committed:
Remove controller interface

saralein committed:
Move callable to new package

saralein commented:
Adjust the auth middleware a bit.  Just gave it a single route for now.

Did some additional adjustments on this PR: https://github.com/saralein/java-server/pull/59.

saralein committed:
Merge pull request #59 from saralein/controllers-to-callables

Controllers to callables

saralein committed:
Merge branch 'master' into auth-middleware
